--- doc/buildman.pe.orig	2018-11-10 08:16:08.000000000 -0700
+++ doc/buildman.pe	2019-01-26 20:35:33.429713783 -0700
@@ -578,7 +578,7 @@ sub make_tex {
 
       SCANMSK: while ( $line = <MSK> ) {
             # treat the {{...}} replacements
-            while ( $line =~ /{{([^}]*)}}/ ) {
+            while ( $line =~ /\{\{([^}]*)\}\}/ ) {
                 $key = $1;
                 if ( $key =~ /date/ ) {
                     $replace=`date +"%d %B %Y"`;
