--- src.x11/bitmaps/checksym.bm.orig	2019-04-15 16:27:54.000000000 -0600
+++ src.x11/bitmaps/checksym.bm	2019-04-24 10:37:25.662595279 -0600
@@ -1,5 +1,5 @@
 #define checksym_width 12
 #define checksym_height 12
-static char checksym_bits[] = {
+static const char checksym_bits[] = {
    0x00, 0x00, 0x00, 0x04, 0x00, 0x06, 0x00, 0x03, 0x80, 0x01, 0xc0, 0x00,
    0x62, 0x00, 0x36, 0x00, 0x1c, 0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00};
--- src.x11/bitmaps/emptymk.bm.orig	2019-04-15 16:27:54.000000000 -0600
+++ src.x11/bitmaps/emptymk.bm	2019-04-24 10:37:25.674595249 -0600
@@ -1,5 +1,5 @@
 #define emptymk_width 12
 #define emptymk_height 12
-static char emptymk_bits[] = {
+static const char emptymk_bits[] = {
    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
--- src.x11/bitmaps/exmark.bm.orig	2019-04-15 16:27:54.000000000 -0600
+++ src.x11/bitmaps/exmark.bm	2019-04-24 10:37:25.675595247 -0600
@@ -1,6 +1,6 @@
 #define exmark_width 32
 #define exmark_height 32
-static char exmark_bits[] = {
+static const char exmark_bits[] = {
    0x00, 0x00, 0x00, 0x00, 0x00, 0x80, 0x00, 0x00, 0x00, 0xc0, 0x01, 0x00,
    0x00, 0x60, 0x03, 0x00, 0x00, 0x60, 0x03, 0x00, 0x00, 0x30, 0x06, 0x00,
    0x00, 0x30, 0x06, 0x00, 0x00, 0x18, 0x0c, 0x00, 0x00, 0x18, 0x0c, 0x00,
--- src.x11/bitmaps/menusym.bm.orig	2019-04-15 16:27:54.000000000 -0600
+++ src.x11/bitmaps/menusym.bm	2019-04-24 10:37:25.675595247 -0600
@@ -1,5 +1,5 @@
 #define menusym_width 12
 #define menusym_height 12
-static char menusym_bits[] = {
+static const char menusym_bits[] = {
    0xfe, 0x03, 0x02, 0x02, 0x02, 0x06, 0x72, 0x06, 0x02, 0x06, 0x72, 0x06,
    0x02, 0x06, 0x72, 0x06, 0x02, 0x06, 0x02, 0x06, 0xfe, 0x07, 0xf8, 0x07};
--- src.x11/bitmaps/return.bm.orig	2019-04-15 16:27:54.000000000 -0600
+++ src.x11/bitmaps/return.bm	2019-04-24 10:37:25.675595247 -0600
@@ -1,5 +1,5 @@
 #define return_width 9
 #define return_height 7
-static char return_bits[] = {
+static const char return_bits[] = {
    0x00, 0x00, 0x44, 0x00, 0x42, 0x00, 0x7d, 0x00, 0x02, 0x00, 0x04, 0x00,
    0x00, 0x00};
--- src.x11/gapgraph.c.orig	2019-04-15 16:27:54.000000000 -0600
+++ src.x11/gapgraph.c	2019-04-24 10:37:25.676595245 -0600
@@ -121,6 +121,7 @@ static void GapGraphDestroy ( Widget w )
 }
 
 
+#if 0
 /****************************************************************************
 **
 *F  GapGraphResize( <w> ) . . . . . . . . . . . . . .  ignore resize requests
@@ -132,6 +133,7 @@ static void GapGraphResize ( Widget w )
     gap->core.width  = gap->gap_graphic.width;
     gap->core.height = gap->gap_graphic.height;
 }
+#endif
 
 
 /****************************************************************************
--- src.x11/pty.c.orig	2019-04-15 16:27:54.000000000 -0600
+++ src.x11/pty.c	2019-04-24 10:37:25.677595242 -0600
@@ -1295,7 +1295,7 @@ int StartGapProcess ( name, argv )
     String          argv[];
 {
     Int             j;       /* loop variables                  */
-    char            c[8];    /* buffer for communication        */
+    char            c[10];   /* buffer for communication        */
     int             master;  /* pipe to GAP                     */
     int             n;       /* return value of 'select'        */
     int             slave;   /* pipe from GAP                   */
