--- src.x11/bitmaps/checksym.bm.orig	2022-02-17 06:50:48.000000000 -0700
+++ src.x11/bitmaps/checksym.bm	2022-02-21 09:29:39.265821138 -0700
@@ -1,5 +1,5 @@
 #define checksym_width 12
 #define checksym_height 12
-static char checksym_bits[] = {
+static const char checksym_bits[] = {
    0x00, 0x00, 0x00, 0x04, 0x00, 0x06, 0x00, 0x03, 0x80, 0x01, 0xc0, 0x00,
    0x62, 0x00, 0x36, 0x00, 0x1c, 0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00};
--- src.x11/bitmaps/emptymk.bm.orig	2022-02-17 06:50:48.000000000 -0700
+++ src.x11/bitmaps/emptymk.bm	2022-02-21 09:29:39.265821138 -0700
@@ -1,5 +1,5 @@
 #define emptymk_width 12
 #define emptymk_height 12
-static char emptymk_bits[] = {
+static const char emptymk_bits[] = {
    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
--- src.x11/bitmaps/exmark.bm.orig	2022-02-17 06:50:48.000000000 -0700
+++ src.x11/bitmaps/exmark.bm	2022-02-21 09:29:39.265821138 -0700
@@ -1,6 +1,6 @@
 #define exmark_width 32
 #define exmark_height 32
-static char exmark_bits[] = {
+static const char exmark_bits[] = {
    0x00, 0x00, 0x00, 0x00, 0x00, 0x80, 0x00, 0x00, 0x00, 0xc0, 0x01, 0x00,
    0x00, 0x60, 0x03, 0x00, 0x00, 0x60, 0x03, 0x00, 0x00, 0x30, 0x06, 0x00,
    0x00, 0x30, 0x06, 0x00, 0x00, 0x18, 0x0c, 0x00, 0x00, 0x18, 0x0c, 0x00,
--- src.x11/bitmaps/menusym.bm.orig	2022-02-17 06:50:48.000000000 -0700
+++ src.x11/bitmaps/menusym.bm	2022-02-21 09:29:39.265821138 -0700
@@ -1,5 +1,5 @@
 #define menusym_width 12
 #define menusym_height 12
-static char menusym_bits[] = {
+static const char menusym_bits[] = {
    0xfe, 0x03, 0x02, 0x02, 0x02, 0x06, 0x72, 0x06, 0x02, 0x06, 0x72, 0x06,
    0x02, 0x06, 0x72, 0x06, 0x02, 0x06, 0x02, 0x06, 0xfe, 0x07, 0xf8, 0x07};
--- src.x11/bitmaps/return.bm.orig	2022-02-17 06:50:48.000000000 -0700
+++ src.x11/bitmaps/return.bm	2022-02-21 09:29:39.266821139 -0700
@@ -1,5 +1,5 @@
 #define return_width 9
 #define return_height 7
-static char return_bits[] = {
+static const char return_bits[] = {
    0x00, 0x00, 0x44, 0x00, 0x42, 0x00, 0x7d, 0x00, 0x02, 0x00, 0x04, 0x00,
    0x00, 0x00};
--- src.x11/gapgraph.c.orig	2022-02-17 06:50:48.000000000 -0700
+++ src.x11/gapgraph.c	2022-02-21 09:29:39.266821139 -0700
@@ -121,6 +121,7 @@ static void GapGraphDestroy ( Widget w )
 }
 
 
+#if 0
 /****************************************************************************
 **
 *F  GapGraphResize( <w> ) . . . . . . . . . . . . . .  ignore resize requests
@@ -132,6 +133,7 @@ static void GapGraphResize ( Widget w )
     gap->core.width  = gap->gap_graphic.width;
     gap->core.height = gap->gap_graphic.height;
 }
+#endif
 
 
 /****************************************************************************
